test failure: panic while executing SQL Logic Test nightly

Found in: https://teamcity.cockroachdb.com/viewLog.html?tab=buildLog&buildTypeId=Cockroach_SqlLogicTests&buildId=530925&_focus=8377

```
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_5.test: 731 statements/queries
E180222 07:41:00.836330 6215 sql/conn_executor.go:496  [n1,client=127.0.0.1:60288,user=root] a SQL panic has occurred while executing "SELECT pk, col0 FROM tab1 WHERE (((col0 > 86) AND (col3 <= 2)) OR ((col0 > 0) AND (col1 > 86.38))) ORDER BY 2 DESC, 1": flow: unexpected 10240 leftover bytes
E180222 07:41:00.836534 6215 util/log/crash_reporting.go:181  [n1,client=127.0.0.1:60288,user=root] a panic has occurred!
E180222 07:41:00.837821 6215 util/log/crash_reporting.go:445  [n1,client=127.0.0.1:60288,user=root] Reported as error 11532ed0c9604c1e9f21a0d3970866c8
panic: flow: unexpected 10240 leftover bytes [recovered]
	panic: panic while executing 1 statements: SELECT _, _ FROM _ WHERE (((_ > _) AND (_ <= _)) OR ((_ > _) AND (_ > _))) ORDER BY _ DESC, _; caused by flow: unexpected 10240 leftover bytes

goroutine 6215 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc422c1b000, 0x219b6c0, 0xc42468ee40, 0x1c30d80, 0xc434384fe0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:510 +0x365
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func2(0xc422c1b000, 0x219b6c0, 0xc42468ee40)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:471 +0x61
panic(0x1c30d80, 0xc434384fe0)
	/usr/local/go/src/runtime/panic.go:505 +0x229
```

--- a/pkg/sql/distsqlrun/sorter.go
+++ b/pkg/sql/distsqlrun/sorter.go
@@ -552,22 +552,26 @@ func (s *sortChunksProcessor) chunkCompleted() (bool, error) {
 }
 
 // fill one chunk of rows from the input and sort them.
-func (s *sortChunksProcessor) fill() (*ProducerMetadata, error) {
+func (s *sortChunksProcessor) fill() error {
 	ctx := s.evalCtx.Ctx()
 
 	var meta *ProducerMetadata
 
-	if s.nextChunkRow == nil {
+	for s.nextChunkRow == nil {
 		s.nextChunkRow, meta = s.input.Next()
-		if meta != nil || s.nextChunkRow == nil {
-			return meta, nil
+		if meta != nil {
+			s.meta = append(s.meta, *meta)
+			continue
+		} else if s.nextChunkRow == nil {
+			return nil
 		}
+		break
 	}
 	s.prefix = s.nextChunkRow
 
 	// Add the chunk
 	if err := s.rows.AddRow(ctx, s.nextChunkRow); err != nil {
-		return nil, err
+		return err
 	}
 
 	defer s.rows.Sort(ctx)
@@ -576,27 +580,30 @@ func (s *sortChunksProcessor) fill() (*ProducerMetadata, error) {
 	// as prefix for the first s.matchLen ordering columns.
 	for {
 		s.nextChunkRow, meta = s.input.Next()
+
 		if meta != nil {
-			return meta, nil
+			s.meta = append(s.meta, *meta)
+			continue
 		}
 		if s.nextChunkRow == nil {
 			break
 		}
 
 		chunkCompleted, err := s.chunkCompleted()
+
 		if err != nil {
-			return nil, err
+			return err
 		}
 		if chunkCompleted {
 			break
 		}
 
 		if err := s.rows.AddRow(ctx, s.nextChunkRow); err != nil {
-			return nil, err
+			return err
 		}
 	}
 
-	return nil, nil
+	return nil
 }
 
 func (s *sortChunksProcessor) Next() (sqlbase.EncDatumRow, *ProducerMetadata) {
@@ -609,12 +616,9 @@ func (s *sortChunksProcessor) Next() (sqlbase.EncDatumRow, *ProducerMetadata) {
 		// If we don't have an active chunk, clear and refill it.
 		if s.rows.Len() == 0 {
 			s.rows.Clear(s.rows.evalCtx.Ctx())
-			meta, err := s.fill()
 			// If that errored or didn't result in an active chunk, we are done.
-			if err != nil || s.rows.Len() == 0 {
+			if err := s.fill(); err != nil || s.rows.Len() == 0 {
 				return nil, s.producerMeta(err)
-			} else if meta != nil {
-				s.meta = append(s.meta, *meta)
 			}
 		}
 

